### PR TITLE
feat: C FFI to verify authorizations

### PR DESCRIPTION
**Pull request by cdata**
This adds C FFI support for verifying that an authorization is still valid, given its CID. The bulk of the implementation here has already shipped, we just didn't have an FFI wrapper for it.

### CODE REVIEW

File: rust/noosphere-core/src/view/sphere.rs [hunk 1]
```diff
@@ -805,13 +805,13 @@ impl<S: BlockStore> Sphere<S> {
...
             if let Some(revocation) = revocations.get(&link).await? {
-                // NOTE: The implication here is that only the sphere itself can
-                // issue revocations The follow-on implicationis that a user
-                // must provide the sphere mnemonic in order to issue a
-                // revocation
-                if revocation.iss != sphere_identity {
-                    warn!("Revocation for {} had an invalid issuer; expected {}, but found {}; skipping...", link, sphere_identity, revocation.iss);
+                // NOTE: The implication here is that only the sphere itself or
+                // the direct issuer may issue revocations
+                if revocation.iss != sphere_identity || revocation.iss != ucan_issuer {
```
Comment (justinabrahms) [inline rust/noosphere-core/src/view/sphere.rs:813]: What's the significance of the ucan issuer? Is that an identity that's allowed to impersonate the sphere_identity?

Comment (jsantell) [reply rust/noosphere-core/src/view/sphere.rs:813]: [Revocation structures](https://github.com/ucan-wg/spec#66-revocation) are related to UCAN, but their own unique structure. In UCAN parlance, the issuer is the signer of the UCAN token (or revocation). Here, we check both that the revocations either came this sphere, or whoever issued the revocation in the first place

File: rust/noosphere/src/ffi/authority.rs [hunk 2]
```diff
@@ -13,6 +13,49 @@ use crate::{
...
+            sphere.verify_authorization(&authorization).await?;
+
+            Ok(()) as Result<_, anyhow::Error>
+        }
+        .await;
+
+        match result {
+            Ok(_) => async_runtime.spawn_blocking(move || callback(context, None)),
+            Err(error) => async_runtime.spawn_blocking(move || {
+                callback(context, Some(NoosphereError::from(error).into()))
```
Comment (jsantell) [inline rust/noosphere/src/ffi/authority.rs:53]: As a failed validation is reported as an error, it'd be nice to have a new error type for authorization/validation failures (to disambiguate from some other e.g. IO failure)

**Review by jsantell — DISMISSED** [review]

File: rust/noosphere/src/ffi/authority.rs [hunk 3]
```diff
@@ -13,6 +13,49 @@ use crate::{
     ffi::{NsError, NsNoosphere, NsSphere},
 };
 
+#[ffi_export]
+/// @memberof ns_sphere_t
+///
+/// For a given authorization CID (given as a base64-encoded string), checks
+/// that the authorization and all of it ancester proofs are valid and have not
```
[the remark this answers is not in the document]
Comment (cdata) [reply rust/noosphere/src/ffi/authority.rs:18]: Should actually be `its` (`it's` is a contraction of `it is`)

**Review by jsantell — DISMISSED** [review]
LGTM!

**Pull request merged**